### ISSUE TITLE
Tags on Catalog Entities

### ISSUE THREAD

**Issue by freben** [spotify/backstage#1997]
## Feature Suggestion

Add the ability to put user defined tags on catalog entities.

These are single-valued strings in essence, to for example classify entities in various ways. This is different to [the labels](https://backstage.io/docs/features/software-catalog/descriptor-format#labels-optional) in `metadata.labels` that are key-value pairs.

## Possible Implementation

We have been discussing back and forth whether tags should be complex and supporting metadata of their own - perhaps colors, description etc. This would probably entail inventing a `Tag` entity kind or similar, that other entities can reference. We have decided against that at least in the initial implementation, in the name of simplicity, until we know more about how the feature will be used in the wild.

Therefore, we propose that tags are simply added as a lightweight optional list of strings to the base shared metadata structure.

Example:

```yaml
metadata:
  name: my-component
  tags:
    - java
    - edge
```

As discussed in #1947, this is compatible with a future extension to the model where these strings could be interpreted as refs (with an implicit `kind` and `namespace`) to a more metadata-rich tag kind. But for now they are just read as-is, as plain strings.

Mockup of how this could look when shown in the Backstage Service Catalog:

![](https://backstage.io/blog/assets/20-05-20/Service_Catalog_MVP.png)

**Comment by Fox32** [comment]
This is something we are interested in. I could offer to implement it (would be at the end of the week).

I would probably split it up into two parts:
* Extend model and show it in the list
* Support the filters on the left side

Do you have a suggestion how it should look on the entity details page?

**Comment by Fox32** [comment]
In addition, I wonder how to handle the casing of the tags. In the example entity the tags are all lower case, in the mockup the start with a capital letter. For the start I would go with no extra handling, but we could extend that later.

**Comment by freben** [comment]
@stefanalund do we have a mockup for the entity details page showing tags?

@Fox32 What you say makes sense. And yeah, a "human readable label" version of the tag, a color, an icon, a description, an owner, a link to documentation about usage - all of those are good candidates to think about for the future. But as you say, we'd probably not process their raw value at all in the initial stage.

**Comment by stefanalund** [comment]
> In addition, I wonder how to handle the casing of the tags. In the example entity the tags are all lower case, in the mockup the start with a capital letter. For the start I would go with no extra handling, but we could extend that later.

@Fox32 personally I think forcing lower case makes sense. 

> Do you have a suggestion how it should look on the entity details page?

@katz95 could you help with this question? Pretty sure we don't have a mockup of this. Feels natural to include it in the "Information" card though. @Fox32 to unblock you, I would suggest to start with just adding it to the Service Catalog table, as outlined in the above mockup.

**Comment by stefanalund** [comment]
Future feature suggestion: make it possible for users to define tags as part creating new components through our [Software Templates](https://backstage.io/docs/features/software-templates/software-templates-index) -- possibly allowing users to choose from existing/used tags presented like a tag "cloud".

**Comment by Fox32** [comment]
> @Fox32 personally I think forcing lower case makes sense.
I will go with the existing `isValidDnsLabel` validation.
